src/cmd/manage_nsfs.js

```diff
@@ -873,8 +881,18 @@ function validate_options_type_by_value(type, input_options_with_data) {
             if ((option === 'name' || option === 'new_name') && (type_of_value === 'number')) {
                 continue;
             }
-            const err_msg = `type of flag ${option} should be ${type_of_option}`;
-            throw_cli_error(ManageCLIError.InvalidArgumentType, err_msg);
+            // special case for allow_bucket_creation
+            if (option === 'allow_bucket_creation' && type_of_value === 'boolean') {
+                continue;
+            }
+            details = `type of flag ${option} should be ${type_of_option}`;
+            throw_cli_error(ManageCLIError.InvalidArgumentType, details);
+        }
+        if (input_options_with_data.allow_bucket_creation &&
```

Comment:
can we do all allow_bucket_creation checks in a single condition?

Reply:
I saw you added `&& check_boolean_string_value(input_options_with_data.show_secrets)` to the other string/boolean options, can't you add it to allow_bucket_creation as well?

Reply:
The `check_boolean_string_value` checks:
1. `allow_bucket_creation` is defined.
2. If it is defined, then we check that the type is `string`.
3. If it is a string we check that it is the strings array that we defined <- here is the change, in the condition we check that it is NOT matched (and add a suitable detail message).

Reply:
You are still throwing InvalidArgument when it's not matching inside the loop. so I still don't see the difference between the checks. If you want a different error I suggested to change  check_boolean_string_value() to throw the errors you want.  
Also, you should check the other options once (check that option is included in an array of the boolean/string options = ['regenerate', 'wide' ...]).

Reply:
Now I understand that you're suggesting adding errors, like we have `InvalidFSBackend` and not using the general `InvalidArgument`. I will add this, thanks.